Lib/tracemalloc.py

```diff
@@ -378,7 +382,7 @@ def _match_traceback(self, traceback):
             return self._match_frame(filename, lineno)
 
     def _match(self, trace):
-        domain, size, traceback = trace
+        domain, size, traceback, length = trace
```

Comment:
I understand that this line breaks if loading from a Python 3.8. Maybe you should add the 4th field in _Traces constructor, if len(trace) == 3, rather than fixing it in the traceback property.

For example:

```
# Convert Python 3.8 trace: add nframe field (set to None)
if traces and len(traces[0]) == 3:
   traces = tuple(trace + (None,) for trace in traces)
```